### PR TITLE
feat: faster ocr

### PR DESCRIPTION
**Pull request by AmineDiro**
- GPU accelerated OCR on onnxruntime
- Docker gpu image
- CI build image docker

### CODE REVIEW

File: libs/megaparse/src/megaparse/api/app.py [hunk 1]
```diff
@@ -114,7 +119,6 @@ async def parse_file(
 
 @app.post(
     "/v1/url",
-    response_model=APIOutputType,
```
Comment (StanGirard) [inline libs/megaparse/src/megaparse/api/app.py:117]: pk ce retrait

Comment (AmineDiro) [reply libs/megaparse/src/megaparse/api/app.py:117]: l'output dans l'api doit etre refaite

File: libs/megaparse/src/megaparse/megaparse.py [hunk 2]
```diff
@@ -1,131 +1,127 @@
...
-        elif file:
+        elif file and file_path is None:
             if not file_extension:
                 raise ValueError(
                     "file_extension should be provided when given file argument"
                 )
             file.seek(0)
-
+        elif file and file_path:
+            raise ValueError("Can't have both a file_path and file")
```
Comment (chloedia) [inline libs/megaparse/src/megaparse/megaparse.py:53]: Repetitive from line 39

File: libs/megaparse/src/megaparse/api/app.py (hunk 1 above)
Comment (Sergii448) [reply libs/megaparse/src/megaparse/api/app.py:117]: с

File: benchmark/process_single_doc.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+import numpy as np
+from megaparse import MegaParse
+from megaparse.parser.unstructured_parser import UnstructuredParser
+from megaparse_sdk.schema.parser_config import StrategyEnum
+
+N_TRY = 1
+
+
+async def process_file(megaparse: MegaParse, file_path: str | Path):
```
Comment (chloedia) [inline benchmark/process_single_doc.py:13]: Question : Is it overkill to do timer decorator in this case ? That we could reuse for other processing

File: benchmark/process_single_doc.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    print(f"All errors : {n_errors}")
+    print(f"Average time taken: {np_list_process_time.mean()}")
+    print(f"Median time taken: {np.median(list_process_time)}")
+    print(f"Standard deviation of time taken: {np.std(list_process_time)}")
+    print(f"Max time taken: {np.max(list_process_time)}")
+    print(f"Min time taken: {np.min(list_process_time)}")
+
+
+if __name__ == "__main__":
+    folder_path = "/Users/amine/data/quivr/parsing/scanned/machine.pdf"
```
Comment (chloedia) [inline benchmark/process_single_doc.py:47]: change path

**Review by chloedia — APPROVED** [review]

**Review by StanGirard — APPROVED** [review]

**Pull request merged**